modules/rules_buf/0.2.0/MODULE.bazel

```diff
@@ -0,0 +1,51 @@
+# Copyright 2021-2023 Buf Technologies, Inc.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#      http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+"Bazel module definition for bzlmod"
+
+module(
+    name = "rules_buf",
+    version = "0.2.0",
+    compatibility_level = 1,
+)
+
+bazel_dep(name = "platforms", version = "0.0.4")
+
+# Only needed because rules_proto doesn't provide the protoc toolchain yet.
+# TODO: remove in the future
+bazel_dep(name = "protobuf", version = "21.7", repo_name = "com_google_protobuf")
+bazel_dep(name = "rules_proto", version = "5.3.0-21.7")
+bazel_dep(name = "rules_go", version = "0.42.0", repo_name = "io_bazel_rules_go")
+bazel_dep(name = "bazel_skylib", version = "1.5.0")
+
+# We depend on gazelle at runtime to generate our proto_library rules
+bazel_dep(name = "gazelle", version = "0.34.0", repo_name = "bazel_gazelle")
+
+go_deps = use_extension("@bazel_gazelle//:extensions.bzl", "go_deps")
+go_deps.from_file(go_mod = "//:go.mod")
+use_repo(
+    go_deps,
+    "com_github_bazelbuild_buildtools",
+    "com_github_stretchr_testify",
+    "in_gopkg_yaml_v3",
+)
+
+# ... and then reach inside to get the gazelle binary from this repository
+non_module_deps = use_extension("@bazel_gazelle//internal/bzlmod:non_module_deps.bzl", "non_module_deps")
```

Comment:
This is only possible because Bazel currently doesn't apply load visibility to module extensions:
https://github.com/bazelbuild/bazel-gazelle/blob/49a501eaee0a7c54950c14981fca5850a201cd09/internal/bzlmod/non_module_deps.bzl#L32

I would highly recommend moving off of this internal interface. If you can explain what this is used for, I can try to offer alternatives.